From 18d9bcfa33939cee345d4d7735bc6081bcc409c8 Mon Sep 17 00:00:00 2001
From: "Baldysiak, Pawel" <pawel.baldysiak@intel.com>
Date: Fri, 30 May 2014 14:38:09 +0000
Subject: [PATCH] Do not set default 'before.layout' when reshaping from RAID4
 to RAID4

Commit fdcad551e9a54c4aa8c4b63160b76e2c539a0441
brings some changes to reshape process.
Setting 'before.layout' when reshaping from RAID4 to another RAID4 is
not really necessary.
If reshape is restarted 'before.layout' will be compared with
'info->array.layout' in reshape_array(). Changes brought by mentioned
commit will cause this comparation return as false, becouse 'array.layout'
is always set to 'ALGORITHM_PARITY_N' in analyse_change() for RAID4, so
reshape will not be continued after reboot/stop.
This patch reverts unnecessary changes.

Signed-off-by: Pawel Baldysiak pawel.baldysiak@intel.com
Reviewed-by: Artur Paszkiewicz <artur.paszkiewicz@intel.com>
Signed-off-by: NeilBrown <neilb@suse.de>
---
 Grow.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Grow.c b/Grow.c
index 2af6c96..2bb4e98 100644
--- a/Grow.c
+++ b/Grow.c
@@ -1305,9 +1305,8 @@ char *analyse_change(char *devname, struct mdinfo *info, struct reshape *re)
 			return "Cannot covert to RAID0 from this level";
 
 		switch (re->level) {
 		case 4:
-			re->before.layout = 0;
 			re->after.layout = 0;
 			break;
 		case 5:
 			re->after.layout = ALGORITHM_PARITY_N;
